x/confidentialtransfers/keeper/keeper.go

```diff
@@ -64,29 +60,36 @@ func NewKeeper(
 	}
 }
 
-func (k BaseKeeper) GetAccount(ctx sdk.Context, address sdk.AccAddress, denom string) (types.Account, bool) {
-	store := ctx.KVStore(k.storeKey)
-	key := types.GetAccountKey(address, denom)
-	if !store.Has(key) {
+func (k BaseKeeper) getAccount(ctx sdk.Context, address sdk.AccAddress, denom string) (types.Account, bool) {
+	ctAccount, found := k.getCtAccount(ctx, address, denom)
+	if !found {
 		return types.Account{}, false
 	}
-
-	var ctAccount types.CtAccount
-	bz := store.Get(key)
-	k.cdc.MustUnmarshal(bz, &ctAccount) // Unmarshal the bytes back into the CtAccount object
 	account, err := ctAccount.FromProto()
 	if err != nil {
 		return types.Account{}, false
 	}
 	return *account, true
 }
 
+func (k BaseKeeper) getCtAccount(ctx sdk.Context, address sdk.AccAddress, denom string) (types.CtAccount, bool) {
```

Comment:
Do we need to export this method as well?

Reply:
For now it's only used in in `getAccount` so I would keep it as is for now.